firebase-appdistribution/src/main/java/com/google/firebase/appdistribution/impl/SignInStorage.java

```diff
@@ -24,48 +24,58 @@
 
 /** Class that handles storage for App Distribution SignIn persistence. */
 class SignInStorage {
-
   private static final String SIGNIN_PREFERENCES_NAME = "FirebaseAppDistributionSignInStorage";
   private static final String SIGNIN_TAG = "firebase_app_distribution_signin";
 
   private final Context applicationContext;
   @Background private final Executor backgroundExecutor;
+  private SharedPreferences sharedPreferences;
+
+  private interface SharedPreferencesFunction<T> {
+    T apply(SharedPreferences sharedPreferences);
+  }
 
   SignInStorage(Context applicationContext, @Background Executor backgroundExecutor) {
     this.applicationContext = applicationContext;
     this.backgroundExecutor = backgroundExecutor;
   }
 
   Task<Void> setSignInStatus(boolean testerSignedIn) {
-    return getSharedPreferences()
-        .onSuccessTask(
-            backgroundExecutor,
-            sharedPreferences -> {
-              sharedPreferences.edit().putBoolean(SIGNIN_TAG, testerSignedIn).apply();
-              return null;
-            });
+    return applyToSharedPreferences(
+        sharedPreferences -> {
+          sharedPreferences.edit().putBoolean(SIGNIN_TAG, testerSignedIn).apply();
+          return null;
+        });
   }
 
   Task<Boolean> getSignInStatus() {
-    return getSharedPreferences()
-        .onSuccessTask(
-            backgroundExecutor,
-            sharedPreferences -> Tasks.forResult(sharedPreferences.getBoolean(SIGNIN_TAG, false)));
+    return applyToSharedPreferences(
+        sharedPreferences -> sharedPreferences.getBoolean(SIGNIN_TAG, false));
   }
 
   boolean getSignInStatusBlocking() {
     return getSharedPreferencesBlocking().getBoolean(SIGNIN_TAG, false);
   }
 
-  private Task<SharedPreferences> getSharedPreferences() {
-    TaskCompletionSource<SharedPreferences> taskCompletionSource = new TaskCompletionSource<>();
-    backgroundExecutor.execute(
-        () -> taskCompletionSource.setResult(getSharedPreferencesBlocking()));
-    return taskCompletionSource.getTask();
-  }
-
   private SharedPreferences getSharedPreferencesBlocking() {
     // This may construct a new SharedPreferences object, which requires storage I/O
     return applicationContext.getSharedPreferences(SIGNIN_PREFERENCES_NAME, Context.MODE_PRIVATE);
   }
+
+  private <T> Task<T> applyToSharedPreferences(SharedPreferencesFunction<T> func) {
+    // Check nullness of sharedPreferences directly even though multiple threads could be calling
+    // this function at once. This isn't a problem because: 1) once it is set it will never be set,
```

Comment:
nit: is this overlong line something `:googleJavaFormat` would fix?
extra-nit: "set back to null" -> "reset back to null"?

Reply:
Huh, it doesn't appear overlong to me (100 characters). `:googleJavaFormat` thinks its fine too.

I like "reset back to null", changed.